drivers/jtag/jtag.c

```diff
@@ -111,22 +111,51 @@ static long jtag_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 
 		if (copy_from_user(&xfer, (const void __user *)arg,
 				   sizeof(struct jtag_xfer)))
+		{
+			dev_err(jtag->miscdev.parent,
+				"JTAG_IOCXFER: copy from user Failed size 0x%lx \n",
+				sizeof(struct jtag_xfer));
 			return -EFAULT;
+		}
 
 		if (xfer.length >= JTAG_MAX_XFER_DATA_LEN)
+		{
+			dev_err(jtag->miscdev.parent,
+				"JTAG_IOCXFER: Error bad length 0x%x \n",
+				xfer.length);
 			return -EINVAL;
+		}
 
 		if (xfer.type > JTAG_SDR_XFER)
+		{
+			dev_err(jtag->miscdev.parent,
+				"JTAG_IOCXFER: Error bad type 0x%x \n",
+				xfer.type);
 			return -EINVAL;
+		}
 
 		if (xfer.direction > JTAG_READ_WRITE_XFER)
+		{
+			dev_err(jtag->miscdev.parent,
+				"JTAG_IOCXFER: Error bad direction 0x%x \n",
+				xfer.direction);
 			return -EINVAL;
+		}
 
 		if (xfer.from > JTAG_STATE_CURRENT)
+		{
+			dev_err(jtag->miscdev.parent,
+				"JTAG_IOCXFER: Error bad from 0x%x \n",
+				xfer.from);
 			return -EINVAL;
-
+		}
```

Comment:
spaces used instead of tab. inconsistent spacing.

Reply:
double check it, do not see any space